keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/Gamma.java

```diff
@@ -40,7 +41,7 @@ public DoubleTensor sample(long[] shape, KeanuRandom random) {
         Tensor.FlattenedView<Double> thetaWrapped = theta.getFlattenedView();
         Tensor.FlattenedView<Double> kWrapped = k.getFlattenedView();
 
-        int length = ArrayUtil.prod(shape);
+        int length = Ints.checkedCast(TensorShape.getLength(shape));
```

Comment:
I've added a `getLengthAsInt()`. The length should actually be a long. The only reason we need to cast it to a int so often is because we create buffers for tensors in java which can only index ints. `getLengthAsInt()` is a bit of a hack for that reason but it's nice to have.